test/e2e/wireguard_test.go

```diff
@@ -127,3 +129,64 @@ func testServiceConnectivity(t *testing.T, data *TestData) {
 	err = data.runNetcatCommandFromTestPod(clientPodName, data.testNamespace, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 	require.NoError(t, err, "Pod %s should be able to connect the service's NodePort %s:%d, but was not able to connect", clientPodName, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 }
+
+// testRoute make sure that the route to Pod on peer Node or route to peer gateway is targeting the WireGuard device.
+func testRoute(t *testing.T, data *TestData) {
+	srcPod, err := data.getAntreaPodOnNode(nodeName(0))
+	require.NoError(t, err)
+	var srcIP, peerGatewayIP, peerPodIP string
+	IPv4, IPv6 := nodeGatewayIPs(0)
+	if IPv4 != "" {
+		srcIP = IPv4
+	} else {
+		srcIP = IPv6
+	}
+	IPv4, IPv6 = nodeGatewayIPs(1)
+	if IPv4 != "" {
+		peerGatewayIP = IPv4
+	} else {
+		peerGatewayIP = IPv6
+	}
+
+	peerPod := randName("agnhost-")
+	require.NoError(t, data.createAgnhostPodOnNode(peerPod, data.testNamespace, nodeName(1), false))
+	defer data.deletePodAndWait(defaultTimeout, peerPod, data.testNamespace)
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, peerPod, data.testNamespace))
+
+	peerPodIPs, err := data.podWaitForIPs(defaultTimeout, peerPod, data.testNamespace)
+	require.NoError(t, err)
+	if peerPodIPs.ipv4 != nil {
+		peerPodIP = peerPodIPs.ipv4.String()
+	} else {
+		peerPodIP = peerPodIPs.ipv6.String()
+	}
+
+	tests := []struct {
```

Comment:
can we just append the verification part to `testPodConnectivity`? it's logically suitable to check connections and routes in that test, and can avoid spending times to create new Pods just to check routes separately.

Reply:
done